ethereum/eip712/domain.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Evmos Foundation
+// This file is part of the Evmos Network packages.
+//
+// Evmos is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Evmos packages are distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Evmos packages. If not, see https://github.com/evmos/evmos/blob/main/LICENSE
+package eip712
+
+import (
+	"strconv"
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+// createEIP712Domain creates the typed data domain for the given chainID.
+func createEIP712Domain(chainID uint64) (apitypes.TypedDataDomain, error) {
+	chainIDAsInt64, err := strconv.ParseInt(strconv.FormatUint(chainID, 10), 10, 64)
```

Comment:
Do we have a util to convert `uint64` to `int64`? If not, it might be useful to create one.

Reply:
`int64(chainID)`

Reply:
But `int64(...)` is susceptible to overflow. In this case, it probably wouldn't be an issue, but in other cases it could be important.

Reply:
we're only using `9000` and `9001` on this repo now so it's fine 👍